crypto/x509/x_x509.c

```diff
@@ -101,6 +101,14 @@ static int x509_cb(int operation, ASN1_VALUE **pval, const ASN1_ITEM *it,
         ASN1_OCTET_STRING_free(ret->distinguishing_id);
         break;
 
+    case ASN1_OP_DUP_POST:
+        {
+            X509 *new = (X509 *)*pval;
```

Comment:
You do have `ret` already...

Reply:
Yes, and also the type cast for `old` is dispensable here. Done.